### ISSUE TITLE
Out of the box nodejs-build-push-pl the docker image is not correct

### ISSUE THREAD

**Issue by oiricaud** [kabanero-io/kabanero-pipelines#297]
Kabanero 0.6

When using the out of the box pipeline `nodejs-build-push-pl` I get the following error when doing a pipeline-run:

step `build` fails
```
[build-push-task : push] {"level":"info","ts":1586290000.5024323,"logger":"fallback-logger","caller":"logging/config.go:69","msg":"Fetch GitHub commit ID from kodata failed: \"KO_DATA_PATH\" does not exist or is empty"}
[build-push-task : push] OUTPUTS_RESOURCE_DOCKER_IMAGE_URL_LOWERCASE=us.icr.io/devops-gse-poc
[build-push-task : push] Getting image source signatures
[build-push-task : push] Copying blob sha256:a8d84d77167b3afd9da72f434813450dc551554562bad59e27319a64ea92f464
[build-push-task : push] Copying blob sha256:c96c867c99157fd1c949813874c607f4d9d29bff10f8f96ea0ae713546ff4f4b
[build-push-task : push] Copying blob sha256:1295eae54c9d95bd8e2c7f83df2a90ac3923d89ec44231fd49f31e7a934f9656
[build-push-task : push] Copying blob sha256:abae0d45101ce11cfd20ce6792008e0852b04c1c21d6ff22f89b49aa35d4ef74
[build-push-task : push] Copying blob sha256:85f69e555a1b647e78d442738170b5a6925e7141335612ce2c58f0ea18ab828d
[build-push-task : push] Copying blob sha256:754e50ec79ad9af9669fa8f614e92c7ea3dbd4609fcc8bd4fd5d6514186123e5
[build-push-task : push] error copying layers and metadata from "containers-storage:[overlay@/var/lib/containers/storage+/var/run/containers/storage:overlay.imagestore=/var/lib/shared,overlay.mount_program=/usr/bin/fuse-overlayfs,overlay.mountopt=nodev,metacopy=on]us.icr.io/devops-gse-poc:latest" to "docker://us.icr.io/devops-gse-poc:latest": Error writing blob: Error initiating layer upload to /v2/devops-gse-poc/blobs/uploads/ in us.icr.io: errors:
[build-push-task : push] denied: requested access to the resource is denied
[build-push-task : push] unauthorized: authentication required
[build-push-task : push]

failed to get logs for task build-push-task : container step-push has failed  : [{"name":"","digest":"","key":"StartedAt","value":"2020-04-07T20:09:44Z","resourceRef":{}}]
TaskRun nodejs-build-push-pl-run-dksd6-r-rg6pd-build-push-task-xcxl5 has failed
```

Upon inspecting the prior step i.e `imagename-lowercase`

Name | Value
-- | --
docker-image | docker-image-qpqnz
git-source | git-source-qpqnz

inspecting the `docker-image-qpqnz` I get 
Name | Value
-- | --
url | us.icr.io/devops-gse-poc

You can see the tag is missing.

`nodejs-build-push-pl.yaml`

``` yaml
apiVersion: tekton.dev/v1alpha1
kind: Pipeline
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: >
      {"apiVersion":"tekton.dev/v1alpha1","kind":"Pipeline","metadata":{"annotations":{"manifestival":"new"},"name":"nodejs-build-push-pl","namespace":"tools"},"spec":{"resources":[{"name":"git-source","type":"git"},{"name":"docker-image","type":"image"}],"tasks":[{"name":"test","resources":{"inputs":[{"name":"source","resource":"git-source"}]},"taskRef":{"kind":"Task","name":"igc-nodejs-test"}},{"name":"build-push-task","resources":{"inputs":[{"name":"git-source","resource":"git-source"}],"outputs":[{"name":"docker-image","resource":"docker-image"},{"name":"git-source","resource":"git-source"}]},"runAfter":["test"],"taskRef":{"kind":"Task","name":"nodejs-build-push-task"}},{"name":"image-scan-task","resources":{"inputs":[{"name":"git-source","resource":"git-source"},{"name":"docker-image","resource":"docker-image"}]},"runAfter":["build-push-task"],"taskRef":{"kind":"Task","name":"nodejs-image-scan-task"}}]}}
    manifestival: new
  creationTimestamp: '2020-04-07T14:51:37Z'
  generation: 2
  name: nodejs-build-push-pl
  namespace: tools
  ownerReferences:
    - apiVersion: kabanero.io/v1alpha2
      controller: false
      kind: Stack
      name: nodejs
      uid: fde0ca81-b522-43ec-a4dd-db1038b5ffaf
  resourceVersion: '25568979'
  selfLink: /apis/tekton.dev/v1alpha1/namespaces/tools/pipelines/nodejs-build-push-pl
  uid: 85f4def5-5c37-4df1-b70c-203c8cab66c2
spec:
  params:
    - description: The event type
      name: event-type
      type: string
    - description: The event ref
      name: event-ref
      type: string
  resources:
    - name: git-source
      type: git
    - name: docker-image
      type: image
  tasks:
    - name: test
      resources:
        inputs:
            - name: source
              resource: git-source
      taskRef:
        kind: Task
        name: igc-nodejs-test
    - name: build-push-task
      resources:
        inputs:
          - name: git-source
            resource: git-source
        outputs:
          - name: docker-image
            resource: docker-image
          - name: git-source
            resource: git-source
      runAfter:
        - test
      taskRef:
        kind: Task
        name: nodejs-build-push-task
    - name: image-scan-task
      resources:
        inputs:
          - name: git-source
            resource: git-source
          - name: docker-image
            resource: docker-image
      runAfter:
        - build-push-task
      taskRef:
        kind: Task
        name: nodejs-image-scan-task
```

**Comment by kvijai82** [comment]
@oiricaud we are working on a fix for this now and will have a new release soon.  Thanks!

**Comment by oiricaud** [comment]
A workaround was to ensure the trigger template was correct.

The `nodejs-build-push-pl-template` had the wrong value:

``` yaml

spec:
        params:
          - name: url
            value: $(params.webhooks-tekton-docker-registry)
        type: image
```
It should have instead 

``` yaml
spec:
        params:
          - name: url
            value: $(params.webhooks-tekton-docker-registry)/$(params.docker-tag)
        type: image
```

**Comment by kvijai82** [comment]
The fix for this is in the Kabanero 0.6.2 release.
